elastic/kibana
# [ResponseOps][Alerting] Implement a query delay mechanism for Alerting rules

Related to https://github.com/elastic/kibana/issues/167061

## Summary

The first PR of two adding the query delay mechanism. This PR adds the query delay to the rules advanced settings.

<img width="899" alt="Screen Shot 2023-10-03 at 7 28 04 AM" src="https://github.com/elastic/kibana/assets/109488926/1449c51d-966e-41d0-878c-5302096e9277">


### Checklist


### To verify

Configuring the query delay settings
1. Go to the [rules page](http://localhost:5601/app/management/insightsAndAlerting/triggersActions/rules) and select the adv settings button
2. Update the query delay setting and save
3. Open the modal again and verify that your setting was saved

To test the accesses:

1. Create a new role and assign kibana privileges to test read and write access
2. Create a new user and assign the new role to that user
3. Log in as that user and verify you corrrect access to the query delay setting

To test serverless
1. Run kibana with serverless and verify that the query delay defaults to 15 instead of 0
